src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_sdfg.py

```diff
@@ -123,8 +123,8 @@ def get_offset_provider(self, offset: str) -> gtx_common.Connectivity | gtx_comm
         return self.offset_provider[offset]
 
     def get_symbol_type(self, symbol_name: str) -> ts.FieldType | ts.ScalarType:
-        assert symbol_name in self.symbol_types
-        return self.symbol_types[symbol_name]
+        assert symbol_name in self.global_symbols
+        return self.global_symbols[symbol_name]
```

Comment:
First of all love `asserts`, but since `dict.__getitem__` will generate a KeyError when the key is missing, that actually tells you what key was not found using the assert here reduces the information, since it does not tell you that.
So either add description to the `assert`, throw your own even more accurate error or get rid of the assert.